### PR TITLE
.NET 8 migration

### PR DESCRIPTION
**Pull request by SrushtiPasari94**
Migrated to .NET 8 version.
Also added the required PackageVersion in central package management file.
The downloaded code was not in working state. Made it working by adding Package versions.

### CODE REVIEW

**Comment by jamesmontemagno** [comment]
@danroth27  anything else @SrushtiPasari94  would need to do moving this to .NET 8?

**Comment by danroth27** [comment]
> @danroth27 anything else @SrushtiPasari94 would need to do moving this to .NET 8?

It's sufficient to update the target framework package versions, but I'd recommend also updating the app to be a Blazor Web App. Instructions are here: https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio#convert-a-blazor-server-app-into-a-blazor-web-app

File: src/eShopOnBlazor/eShopOnBlazor.csproj [hunk 1]
```diff
@@ -37,10 +37,17 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Autofac" Version="6.5.0" />
-    <PackageReference Include="EntityFramework" Version="6.4.4" />
-    <PackageReference Include="log4net" Version="2.0.15" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Log4Net.AspNetCore" Version="6.1.0" />
+    <PackageReference Include="EntityFramework" />
```
Comment (danroth27) [inline src/eShopOnBlazor/eShopOnBlazor.csproj:40]: I don't recognize this pattern for specifying default package versions. Are we sure this works?

Comment (SrushtiPasari94) [reply src/eShopOnBlazor/eShopOnBlazor.csproj:40]: @danroth27 earlier the repo had this below error 
image

Resolved that error by adding PackageVersion.

Tested. Its working fine with that

**Comment by SrushtiPasari94** [comment]
> > @danroth27 anything else @SrushtiPasari94 would need to do moving this to .NET 8?
> 
> It's sufficient to update the target framework package versions, but I'd recommend also updating the app to be a Blazor Web App. Instructions are here: https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio#convert-a-blazor-server-app-into-a-blazor-web-app

I will start on this.

**Comment by SrushtiPasari94** [comment]
@danroth27 as per suggestion, I have updated the Blazor server app to be a Blazor Web App. Please Review the changes.

cc @jamesmontemagno

**Comment by jamesmontemagno** [comment]
@danroth27  can you give this a quick review?

**Comment by danroth27** [comment]
@jamesmontemagno I'm out until the new year. If you need this reviewed sooner than that, then please work with @mkArtakMSFT to see if someone on the engineering team can take look.

**Comment by jamesmontemagno** [comment]
I was able to pull this down and getting it running on my Windows machine. We will want to probably update the EF system to use SQLite so it works on more machine types and probably update some styles as well to make it look a bit better via css.

**Review by jamesmontemagno — APPROVED** [review]

**Pull request merged**